runtime/vm/resolvesupport.cpp

```diff
@@ -2091,6 +2101,8 @@ resolveOpenJDKInvokeHandle(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cp
 
 	Trc_VM_resolveInvokeHandle_Entry(vmThread, ramCP, cpIndex, resolveFlags);
 
+	Assert_VM_true(J9_ARE_NO_BITS_SET(resolveFlags, J9_RESOLVE_FLAG_NO_CP_UPDATE));
```

Comment:
Why is this specific to `OpenJDKInvokeHandle`? shouldn't it apply to `resolveInvokeDynamic` and `resolveConstantDynamic` as well?

Reply:
None of these are currently used by the `ConstantPool` code. `resolveInvokeDynamic` and `resolveConstantDynamic` already have a similar assertion. The OpenJDK implementation code is using CAS to update the cache, and potentially returning the old contents, so I thought it best to not to unnecessarily change the code. The assertion will prevent bugs creeping in in the future if this is eventually called differently.